test/micro/org/openjdk/bench/java/lang/MergeStoreBench.java

```diff
@@ -0,0 +1,870 @@
+            UNSAFE.putLong(bytes8, Unsafe.ARRAY_BYTE_BASE_OFFSET + i * 8, v);
+            sum += v;
+        }
+        BH.consume(sum);
+    }
+
+    @Benchmark
+    public void setLongU(Blackhole BH) {
+        long sum = 0;
+        for (int i = 0; i < longs.length; i++) {
+            long v = longs[i];
+            UNSAFE.putLong(bytes8, Unsafe.ARRAY_BYTE_BASE_OFFSET + i * 8, v);
+            sum += v;
+        }
+        BH.consume(sum);
+    }
+
+    @Benchmark
+    public void putChars4(Blackhole BH) {
+        long sum = 0;
+        for (int i = 0; i < longs.length; i++) {
+            putChars4(bytes8, i * 4, 'n', 'u', 'l', 'l');
+            sum += longs[i];
+        }
+        BH.consume(sum);
+    }
+
+    @Benchmark
+    public void putChars4UB(Blackhole BH) {
+        long sum = 0;
+        for (int i = 0; i < longs.length; i++) {
+            putChars4UB(bytes8, i * 4, 'n', 'u', 'l', 'l');
+            sum += longs[i];
+        }
+        BH.consume(sum);
+    }
+
+    @Benchmark
+    public void putChars4UC(Blackhole BH) {
+        long sum = 0;
+        for (int i = 0; i < longs.length; i++) {
+            putChars4UC(bytes8, i * 4, 'n', 'u', 'l', 'l');
+            sum += longs[i];
+        }
+        BH.consume(sum);
+    }
+
+    static int getIntB(byte[] array, int offset) {
+        return ((array[offset    ] & 0xff) << 24)
+             | ((array[offset + 1] & 0xff) << 16)
+             | ((array[offset + 2] & 0xff) <<  8)
+             | ((array[offset + 3] & 0xff)      );
+    }
+
+    static int getIntBU(byte[] array, int offset) {
+        final long address = Unsafe.ARRAY_BYTE_BASE_OFFSET + offset;
+        return ((UNSAFE.getByte(array, address    ) & 0xff) << 24)
+             | ((UNSAFE.getByte(array, address + 1) & 0xff) << 16)
+             | ((UNSAFE.getByte(array, address + 2) & 0xff) <<  8)
+             | ((UNSAFE.getByte(array, address + 3) & 0xff)      );
+    }
+
+    static int getIntL(byte[] array, int offset) {
+        return ((array[offset       ] & 0xff)      )
+                | ((array[offset + 1] & 0xff) <<  8)
+                | ((array[offset + 2] & 0xff) << 16)
+                | ((array[offset + 3] & 0xff) << 24);
+    }
+
+    static int getIntRB(byte[] array, int offset) {
+        return Integer.reverseBytes(getIntB(array, offset));
+    }
+
+    static int getIntRBU(byte[] array, int offset) {
+        return Integer.reverseBytes(getIntBU(array, offset));
+    }
+
+    static int getIntRL(byte[] array, int offset) {
+        return Integer.reverseBytes(getIntL(array, offset));
+    }
+
+    static int getIntRLU(byte[] array, int offset) {
+        return Integer.reverseBytes(getIntLU(array, offset));
+    }
+
+    static void setIntB(byte[] array, int offset, int value) {
+        array[offset    ] = (byte) (value >> 24);
+        array[offset + 1] = (byte) (value >> 16);
+        array[offset + 2] = (byte) (value >>  8);
+        array[offset + 3] = (byte) (value      );
+    }
```

Comment:
Big endian is often used in network data transmission scenarios, and it is common to process big endian data on a little endian machine. In this case, can it be optimized to Integer.reverseBytes & putInt on a LittleEndian machine? `setIntB -> setIntRL`

Reply:
I had already suggested that here:
https://github.com/openjdk/jdk/pull/19218#pullrequestreview-2076196539
Feel free to file an RFE. Maybe someone wants to work on it. I think it would not be that hard to make it work given all the code that is already there now. And it would be helpful in for both big/little endian to be able to do both orders.